packages/authentication/test/unit/providers/auth-metadata.provider.unit.ts

```diff
@@ -65,6 +65,17 @@ describe('AuthMetadataProvider', () => {
         );
         expect(authMetadata).to.be.undefined();
       });
+
+      it('returns undefined when the class or method is missing', async () => {
```

Comment:
Can we separate this test into two? i.e. one for class missing and the other class defined but method missing? or is that not useful?

Reply:
@b-admike The method lives in the controller, so if the controller class is missing then the method will definitely be missing. Which means the only valid scenario is no controller class + no method.

What we could improve in the next PR is the scenario that Miroslav brings in his comment:

```ts
const spec: OperationObject = { /*...*/ };
function greet(name: string) {
  return `hello ${name}`;
}

app.route('get', '/', spec, greet);
``` 
And as Raymond said, the auth metadata could be included in the OpenAPI spec.